transport/http/transport.go

```diff
@@ -174,16 +217,27 @@ func (a *Transport) RetainPeer(pid peer.Identifier, sub peer.Subscriber) (peer.P
 	return p, nil
 }
 
+// for httpPeer to obtain a peer or nil
+func (a *Transport) getPeerWithLock(addr string) *httpPeer {
+	a.lock.Lock()
+	defer a.lock.Unlock()
+
+	if p, ok := a.peers[addr]; ok {
+		return p
+	}
+	return nil
+}
+
 // **NOTE** should only be called while the lock write mutex is acquired
-func (a *Transport) getOrCreatePeer(pid hostport.PeerIdentifier) *hostport.Peer {
+func (a *Transport) getOrCreatePeer(pid hostport.PeerIdentifier) *httpPeer {
 	if p, ok := a.peers[pid.Identifier()]; ok {
 		return p
 	}
 
-	p := hostport.NewPeer(pid, a)
-	p.SetStatus(peer.Available)
-
+	p := newPeer(pid, a)
 	a.peers[p.Identifier()] = p
+	a.connectorsGroup.Add(1)
+	go p.maintainConn()
 
 	return p
```

Comment:
Consumers of this method can cope with the new peer not being immediately available, right?